lib/array_intersection.rb

```diff
@@ -1,4 +1,38 @@
 def intersection(list1, list2)
-  raise NotImplementedError, "Intersection not implemented"
+  def intersection(array1, array2)
```

Comment:
You have `def intersection` inside an intersection method

Other than that, this works, but it could use some cleaning up.